native/exe/ServiceControlLaunch/ServiceControlLaunch.cpp

```diff
@@ -14,29 +14,31 @@
  * limitations under the License.
  */
 // ServiceControlLaunch.cpp : Defines the entry point for the application.
-//
 
-#include "stdafx.h"
-
-
-// ServiceControlLauncher.cpp : Defines the entry point for the console application.
-//
+/*
+*		*********************************************************
+*		Microsoft Visual Studio project codepage set to MBCS 
+*		as JAVA does not support being passed Unicode parameters
+*		*********************************************************
+*/
 
 #include "stdafx.h"
 
-
-/*
- * Copyright 2001-2004 Frey Technologies, LLC. All rights reserved.
- */
 #include <windows.h>
 #include <stdlib.h>
 #include <jni.h>
 #include <stdio.h>
 #include <direct.h>
 
 
-#define JVM_MISSING MessageBox(NULL, "Could not get information on current JVM.\nPlease install Java Runtime Environment 1.4","Java Missing", MB_OK);\
+#define JVM_MISSING MessageBox(NULL, "Could not get information on current JVM.\nPlease install Java Runtime Environment 1.7 or higher (Java 1.8 preferred)", "Java Missing", MB_OK);\
 
+#ifndef _MSC_VER // Cater for compiling not in MSVC

```

Comment:
This is not good practice to redefine 'safe' string functions to be their non-safe equivalent. There's no reason I can think this file wouldn't be compiled with MSVC either since it's making a Windows EXE.

Reply:
Removed